src/modules/ekf2/EKF/covariance.cpp

```diff
@@ -530,17 +530,16 @@ void Ekf::resetQuatCov()
 {
 	zeroQuatCov();
 
-	// define the initial angle uncertainty as variances for a rotation vector
-	Vector3f rot_vec_var;
-	rot_vec_var.setAll(sq(_params.initial_tilt_err));
-
-	initialiseQuatCovariances(rot_vec_var);
+	const float tilt_var = sq(fmaxf(_params.initial_tilt_err, 1.0e-2f));
+	Vector3f rot_var_ned(tilt_var, tilt_var, 0.f);
 
 	// update the yaw angle variance using the variance of the measurement
 	if (_params.mag_fusion_type != MagFuseType::NONE) {
 		// using magnetic heading tuning parameter
-		increaseQuatYawErrVariance(sq(fmaxf(_params.mag_heading_noise, 1.0e-2f)));
+		rot_var_ned(2) = (sq(fmaxf(_params.mag_heading_noise, 1.0e-2f)));
```

Comment:
Do we still need this? If/when mag heading starts the first time it's going to resetQuatStateYaw() which calls increaseQuatYawErrVariance().

Reply:
Right, since we anyway increase it when the fusion starts, we could remove it from here

Reply:
But `resetQuatCov` is also called here: https://github.com/PX4/PX4-Autopilot/blob/ecbc5b83a0012b7ca2ca8bb79a694718d70799c5/src/modules/ekf2/EKF/mag_fusion.cpp#L79

Reply:
Make it an argument with a default value of 0 or NAN and then check if it's been provided?